src/components/Attachments/AttachmentCarousel/index.js

```diff
@@ -49,6 +49,10 @@ function AttachmentCarousel({report, reportActions, parentReportActions, source,
 
         const initialPage = _.findIndex(attachmentsFromReport, compareImage);
 
+        if (_.isEqual(attachments, attachmentsFromReport)) {
+            return;
+        }
```

Comment:
This early return caused the state variable `page` to be stale (at `0`) where it should have been updated to `-1` if the `attachments` array is empty